lib/facil/fio.c

```diff
@@ -2313,7 +2313,11 @@ int fio_set_non_block(int fd) {
   if (-1 == (flags = fcntl(fd, F_GETFL, 0)))
     flags = 0;
   // printf("flags initial value was %d\n", flags);
-  return fcntl(fd, F_SETFL, flags | O_NONBLOCK | O_CLOEXEC);
+  return fcntl(fd, F_SETFL, flags | O_NONBLOCK
```

Comment:
This isn't very readable... maybe repeat the whole line in the different `#if` sections rather then just repeat the flag?